### ISSUE TITLE
UX Polish

### ISSUE THREAD

**Issue by davidflanagan** [mdn/sprints#1588]
# User story
As a MDN frontend engineer, I want our beta pages to look great and work well.

This user story is for miscellaneous UX polish and fixing minor paper cut issues on the beta pages. Nick is taking a look at the beta pages and will probably have requests for changes. Kadir has highlighted some issues at https://miro.com/app/board/o9J_kx-Gscg=/

I can't estimate this story until I get feedback from Nick...

# Acceptance criteria

- Kadir and Nick are happy enough with the look-and-feel of the beta pages.

# Tasks
- [x] Put content in containers with a max width ~ 1400px (as in current site)
- [x] Update h3 to not have full width background
- [x] Update styling of main navigation
- [x] Update styling of main nav - dropdown
- [x] Update styling of search bar
- [x] Add anchor links to headings/sections
- [x] [logged in users] Move edit link into title area as in design below
- [x] [logged in users] Add history link (this a content related link Kadir wanted to keep)
- [x] [logged in users] Adjust styling of avatar in header (we may want to update the default avatar as well)
- [x] Move Languages dropdown to right side of breadcrumbs bar, 
- [x]  Display the current language instead of the word "Languages" in the translation menu
- [x] Remove github logo from sign in link (if we are moving towards having multiple options, such as using google accounts and custom MDN, this will no longer make sense in the future)
- [x] Remove black Jump Bar (will be putting together a design for how we will handle still having these links as a table of contents in the sidebar for the learning area section of the site)
- [x] Move breadcrumbs to essentially where Jump Bar was, and update styling
- [x] Implement loading indicator (top bar) - _will provide design_

# Design

Mobile - phone: _in progress_
Mobile - tablet (768+): https://zpl.io/bzqDggz
Desktop (1024+): https://zpl.io/2EgYNMx
Large Screen (1440+): https://zpl.io/2j5BG0O
XLarge Screen - _showing max width_ (1600): https://zpl.io/b64YrXm

Not Signed In: https://zpl.io/bo8rdXZ
Contributor (with content links): https://zpl.io/2yqpnZo

Main Nav - Target Area (showing padding of main list items): https://zpl.io/agQyNdZ
Main Nav - Dropdown: https://zpl.io/VDARndv
Main Nav - Dropdown - Item Hover: https://zpl.io/Vx0olXg

**Loading Indicator**
Initial (content already loaded): https://zpl.io/amdreMQ
Loading - 1/3 finished: https://zpl.io/a7WnQdj
Loading - 2/3 finished: https://zpl.io/aX3o95p
_loading bar would slide up/fade out when finished, similar to examples below_
http://nanobar.jacoborus.codes/
http://ricostacruz.com/nprogress/

**Table of Contents**
https://zpl.io/aw6LAl1
https://zpl.io/2pMyYqj (long list item example)

**Comment by peterbe** [comment]
Should the section navigating bar thingy that links to the sub-sections be part of this?

**Comment by davidflanagan** [comment]
@peterbe: the scope is the entire UX of the beta site, so if you've got changes to suggest to or bugs to report about that navigation bar, feel free to suggest them. Keep in mind, however that Nick is finalizing a design, and that I'll defer to him on any design issues.

**Comment by nickbrandt** [comment]
@davidflanagan I've included some design links in the issue description, as well as added some tasks that represent some of those changes (as well as things we discussed during our meeting). It's a good amount of work, though some of those items hopefully will be fairly simple to implement.

One of these items involves removing the current Jump Bar, which has some implications for some of our longer pages in the learning area. @atopal and I talked about doing something different for that section, where we can retain those anchor links, but put them in the sidebar as more of a table of contents. I still need to put together a design for that.

@atopal can you take a look at the designs as well as the tasks, and let @davidflanagan and I know what ones you see as priority, vs. what you think should be broken out into a separate story (future enhancement)?

Let me know if you have any questions about this @davidflanagan , I'd be happy to jump on a call to talk though any of this!

**Comment by atopal** [comment]
@nickbrandt I can't see the designs, can you send me an invite link for zeplin?

I reordered the tasks for priority. I also think that the last 3 should be handled separately (removing github logo, jump bar, breadcrumbs), and this issue should be limited to the minimum we need to get done before we can switch over.

**Comment by nickbrandt** [comment]
@atopal you should have access now. @davidflanagan the values should now reflect actual size in Zeplin, it was indeed a setting that was wrong.

**Comment by nickbrandt** [comment]
@atopal I honestly think with the launch of the updated site in React, this is the best time to move the breadcrumbs to a logical place (not in the sidebar). I think it is also the best time to remove the Jump Bar, but if you want to hold off on that for now, I can at least put together a design where the Jump Bar remains, but we still move the breadcrumbs out of the sidebar. I understand if it still makes sense to break that out into a separate issue though. Let me know your thoughts on this.

**Comment by nickbrandt** [comment]
@davidflanagan I am also curious about when we plan to address page transitions, specifically giving users an affordance that the page is loading (from our convo here: https://github.com/mozilla/kuma/pull/5373#issuecomment-488277276). As I understand it now, when a user clicks on various links within MDN on the beta site, there is no indication that the page is loading (granted as long as the connection is good, its very quick). Though for users with a poor connection, this could be problematic. Has this work been prioritized somewhere?

**Comment by davidflanagan** [comment]
@nickbrandt: I see the updated sizes. Thanks. 
Yes, let's do page transitions as part of this work as well.

@atopal: the three items you're proposing to move should all be pretty easy to do, so unless you're worried that jump bar fans will revolt or something, I'd like to just go ahead and do it.

**Comment by davidflanagan** [comment]
Will emailed this list of issues he's noticed. This may not be the right user story to put them in, but I want to record them somewhere:

* no syntax highlighting
* "jump to" is not sticky
* section anchors are not there
* browser names in compat tables disappear at certain widths (screenshot attached)
* no "Open in..." buttons on live samples
* CSS interactive example responsiveness: the CSS examples are supposed to switch into top/bottom display in narrower windows

**Comment by davidflanagan** [comment]
I've begun a pull request here: https://github.com/mozilla/kuma/pull/5442

Currently it has the new search box design and header design changes. There is more to come, and I'll keep pushing commits to that PR for a while.

**Comment by davidflanagan** [comment]
https://github.com/mozilla/kuma/pull/5442 is now ready for review: it includes the header, titlebar and breadcrumbs changes. (But not the menu changes yet)

https://github.com/mozilla/kuma/pull/5448 is another PR ready for review that adds anchor links to section headings and fixes the too-wide black backgrounds for h3 elements.

**Comment by davidflanagan** [comment]
I've implemented the 1400px max width. Here's what it looks like with a 1600px wide window:

<img width="1607" alt="Screen Shot 2019-05-28 at 1 45 06 PM" src="https://user-images.githubusercontent.com/516914/58511184-271b2280-814f-11e9-9f78-b2d15f56eafe.png">

**Comment by nickbrandt** [comment]
@davidflanagan I've updated the designs in the description with some additions for the main menu.

In whether or not we revert this to on hover or on click, I will leave that up to you. I know you already did work to implement it on click. However, if we do keep in on click, we will need to make it possible to get to those main nav links (as we had discussed previously). Currently, clicking on those items just toggles the menu. So we would need to add those links as the first list item in the dropdown or something.

The "target area" design simply shows what we could set the padding area for the main list items, as to not cause the dropdowns to appear unintentionally (which was one of your concerns). The dotted lines would not actually be visible, just using that to indicate the reduced padding area vs the current site.

The "menu-hover" design shows what would happen when you then hover over one of the list items within the dropdown (adding the blue background and bolding the text).

If you have any other questions regarding the menu, just let me know!

**Comment by nickbrandt** [comment]
@davidflanagan I've provided some designs in the description for the loading indicator.

As I looked into this a bit more, I started thinking that we should just lose the current top bar (or reuse this) so that you only see it when the page is loading. The idea here is once the content is fully loaded, the bar would slide up/fade out quickly. This bar would go overtop the content (and i reduced the height a bit), so the page doesn't jump when the bar goes away.

I also played with an option where essentially the current gradient bar is gray and its background progresses to the gradient as the page loads, keeping the gradient once finished (as seen below). However, I think it is more clear to users that the page has loaded to simply lose the bar once finished vs keeping it on the page. 

You can see this in practice in the examples I included in the description, as well as sites like youtube.com and github. 

Let me know if you have any thoughts or questions on this.

![Screen Shot 2019-05-29 at 2 16 36 PM](https://user-images.githubusercontent.com/5185090/58588210-65cade80-821c-11e9-9a15-1ad8cdf31664.png)
![Screen Shot 2019-05-29 at 2 16 08 PM](https://user-images.githubusercontent.com/5185090/58588211-65cade80-821c-11e9-8027-1aaef0147c90.png)

**Comment by davidflanagan** [comment]
https://github.com/mozilla/kuma/pull/5463 changes the language menu to display the name of the current language instead of the icon. It also just removes the menu in cases where there are no translations available.

**Comment by atopal** [comment]
For context: the gradient bar was added when users complained that MDN had gotten too black and white after the last redesign. I'm not too attached to it.

**Comment by jmswisher** [comment]
> [mozilla/kuma#5463](https://github.com/mozilla/kuma/pull/5463) changes the language menu to display the name of the current language instead of the icon. It also just removes the menu in cases where there are no translations available.

The language menu disappearing is just for the read-only domain, right? I hope that the "Add a translation" menuitem still exists somewhere on the wiki domain.

**Comment by davidflanagan** [comment]
@jmswisher: yes, the wiki domain is untouched, so it will still be there. To be honest, I'd forgotten about the 'add a translation" menu item.It might be reasonable to put that back in in the case where the user is a logged-in contributor.

**Comment by davidflanagan** [comment]
I'm calling this user story done. I'm sure we'll need another round of touch-ups, but the UX now matches what Nick spec'ed out on Zeplin
